components/org.wso2.carbon.identity.sso.agent/src/main/java/org/wso2/carbon/identity/sso/agent/saml/SAML2SSOManager.java

```diff
@@ -698,12 +698,16 @@ protected String encodeRequestMessage(SignableSAMLObject requestMessage, String
             for (AttributeStatement statement : attributeStatementList) {
                 List<Attribute> attributesList = statement.getAttributes();
                 for (Attribute attribute : attributesList) {
-                    List<String> valueList = new ArrayList<>();
-                    for (XMLObject xmlObject : attribute.getAttributeValues()) {
-                        valueList.add(xmlObject.getDOM().getTextContent());
+                    List<XMLObject> multipleAttributeValues = attribute.getAttributeValues();
+                    if (!multipleAttributeValues.isEmpty()) {
+                        List<String> valueList = new ArrayList<String>();
+                        for (XMLObject attributeVal : multipleAttributeValues) {
+                            Element value = attributeVal.getDOM();
+                            valueList.add(value.getTextContent());
+                        }
+                        String attributeValue = StringUtils.join(valueList.iterator(), ",");
```

Comment:
This adds the value to the list even if there is no value . ie empty right ?. Is that the expected behaviour ?